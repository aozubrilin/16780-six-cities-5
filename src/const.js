export const STAR_TO_PERSENT = 20;

export const OfferCardClass = {
  MAIN: `cities`,
  NEAR: `near-places`,
  FAVORITE: `favorites`,
};

export const CITIES = [
  `Paris`,
  `Cologne`,
  `Brussels`,
  `Amsterdam`,
  `Hamburg`,
  `Dusseldorf`
];

export const SortType = {
  POPULAR: `Popular`,
  TO_HIGHT: `Price: low to high`,
  TO_LOW: `Price: high to low`,
  TOP_RATE: `Top rated first`,
};
