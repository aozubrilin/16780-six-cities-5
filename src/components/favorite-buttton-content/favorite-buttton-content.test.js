import React from "react";
import renderer from "react-test-renderer";
import FavoriteButtonContent from "./favorite-buttton-content";
import {favoriteButtonProperty} from "../../const";


describe(`FavoriteButtonContent snapshot`, () => {

  it(`FavoriteButtonContent render correctly with isFavorite={false} and buttonProperty={favoriteButtonProperty.offerCard}`, () => {
    const tree = renderer.create(
        <FavoriteButtonContent
          isFavorite={false}
          buttonProperty={favoriteButtonProperty.offerCard}
        />
    )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`FavoriteButton render correctly with isFavorite={true} and buttonProperty={favoriteButtonProperty.offerCard}`, () => {
    const tree = renderer.create(
        <FavoriteButtonContent
          isFavorite={true}
          buttonProperty={favoriteButtonProperty.offerCard}
        />
    )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`FavoriteButton render correctly with isFavorite={false} and buttonProperty={favoriteButtonProperty.propertyCard}`, () => {
    const tree = renderer.create(
        <FavoriteButtonContent
          isFavorite={false}
          buttonProperty={favoriteButtonProperty.propertyCard}
        />
    )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`FavoriteButton render correctly with isFavorite={true} and buttonProperty={favoriteButtonProperty.propertyCard}`, () => {
    const tree = renderer.create(
        <FavoriteButtonContent
          isFavorite={true}
          buttonProperty={favoriteButtonProperty.propertyCard}
        />
    )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
