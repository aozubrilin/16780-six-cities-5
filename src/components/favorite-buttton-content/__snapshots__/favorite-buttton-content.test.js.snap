// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FavoriteButtonContent snapshot FavoriteButton render correctly with isFavorite={false} and buttonProperty={favoriteButtonProperty.propertyCard} 1`] = `
Array [
  <svg
    className="property__bookmark-icon"
    height="33"
    width="31"
  >
    <use
      xlinkHref="#icon-bookmark"
    />
  </svg>,
  <span
    className="visually-hidden"
  >
    To
     bookmarks
  </span>,
]
`;

exports[`FavoriteButtonContent snapshot FavoriteButton render correctly with isFavorite={true} and buttonProperty={favoriteButtonProperty.offerCard} 1`] = `
Array [
  <svg
    className="place-card__bookmark-icon"
    height="19"
    width="18"
  >
    <use
      xlinkHref="#icon-bookmark"
    />
  </svg>,
  <span
    className="visually-hidden"
  >
    In
     bookmarks
  </span>,
]
`;

exports[`FavoriteButtonContent snapshot FavoriteButton render correctly with isFavorite={true} and buttonProperty={favoriteButtonProperty.propertyCard} 1`] = `
Array [
  <svg
    className="property__bookmark-icon"
    height="33"
    width="31"
  >
    <use
      xlinkHref="#icon-bookmark"
    />
  </svg>,
  <span
    className="visually-hidden"
  >
    In
     bookmarks
  </span>,
]
`;

exports[`FavoriteButtonContent snapshot FavoriteButtonContent render correctly with isFavorite={false} and buttonProperty={favoriteButtonProperty.offerCard} 1`] = `
Array [
  <svg
    className="place-card__bookmark-icon"
    height="19"
    width="18"
  >
    <use
      xlinkHref="#icon-bookmark"
    />
  </svg>,
  <span
    className="visually-hidden"
  >
    To
     bookmarks
  </span>,
]
`;
